### PR TITLE
(#2190228) Support defining mount units on the kernel command line

### PR DESCRIPTION
**Pull request by dtardon**
Resolves: [#2190228](https://bugzilla.redhat.com/show_bug.cgi?id=2190228)

### CODE REVIEW

File: src/fstab-generator/fstab-generator.c [hunk 1]
```diff
@@ -962,6 +1118,8 @@ int main(int argc, char *argv[]) {
                 }
         }
 
+        r = add_mounts_from_cmdline();
```
Comment (msekletar) [inline src/fstab-generator/fstab-generator.c:1122]: I think the error handling is incorrect here. Main error code returned by generator depends on value of `r`. Previously, generator would return error return code when `fstab` processing failed. Now, (w/o any units on the command line which is the default case) this potential error would be obscured by `0` returned from `add_mounts_from_cmdline()` when there are no mounts on the command line.

**Review by msekletar — CHANGES_REQUESTED** [review]

File: src/fstab-generator/fstab-generator.c [hunk 2]
```diff
@@ -885,41 +1083,46 @@
 
         /* Always honour root= and usr= in the kernel command line if we are in an initrd */
         if (in_initrd()) {
-                int k;
-
                 r = add_sysroot_mount();
+                if (r < 0 && ret >= 0)
```
[the remark this answers is not in the document]
Comment (dtardon) [reply src/fstab-generator/fstab-generator.c:1087]: True, but it's harmless.

**Comment by jamacku** [comment]
@dtardon Could you please have a look at the conflict in `fstab-generator.c`. Thank you

Also please reference the [RHEL-5989](https://issues.redhat.com/browse/RHEL-5989) instead of [#2190228](https://bugzilla.redhat.com/show_bug.cgi?id=2190228)

**Review by jamacku — CHANGES_REQUESTED** [review]
Related upstream follow-ups:
- https://github.com/systemd/systemd/pull/28518
- https://github.com/systemd/systemd/pull/28544

We should include them as well.

**Comment by jamacku** [comment]
It's too late for RHEL 8, hence closing this PR.

**Pull request closed without merging**